src/ripple/nodestore/impl/DatabaseShardImp.cpp

```diff
@@ -1074,19 +1143,59 @@ DatabaseShardImp::initConfig(std::lock_guard<std::mutex>&)
         return fail("'path' missing");
 
     {
-        std::uint64_t sz;
-        if (!get_if_exists<std::uint64_t>(section, "max_size_gb", sz))
-            return fail("'max_size_gb' missing");
+        get_if_exists(section, "max_historical_shards", maxHistoricalShards_);
+
+        Section const& historicalShardPaths =
+            config.section(SECTION_HISTORICAL_SHARD_PATHS);
 
-        if ((sz << 30) < sz)
-            return fail("'max_size_gb' overflow");
+        auto values = historicalShardPaths.values();
 
-        // Minimum storage space required (in gigabytes)
-        if (sz < 10)
-            return fail("'max_size_gb' must be at least 10");
+        std::sort(values.begin(), values.end());
+        values.erase(std::unique(values.begin(), values.end()), values.end());
 
-        // Convert to bytes
-        maxFileSz_ = sz << 30;
+#if BOOST_OS_LINUX
```

Comment:
Done

Reply:
That's great, thanks. One minor change I'd still like: instead of one function with the preprossor directives inside the function, can we change it so there are two separate functions and the preprossor directives control which function is compiled in?

Reply:
I missed this comment. What's the benefit vs having the directives inside the function?